app/routes/shop.py

```diff
@@ -146,4 +148,54 @@ def upload(id):
   setattr(dive_shop, 'logo_img', s3_url)
   db.session.commit()
 
-  return { 'msg': 'dive shop successfully updated' }
\ No newline at end of file
+  return { 'msg': 'dive shop successfully updated' }
+
+@bp.route('/typeahead')
+@cache.cached(query_string=True)
+def get_typeahead():
+  query = request.args.get('query')
+  results = []
+  dive_shops_by_name = DiveShop.query \
+    .filter(DiveShop.name.ilike('%'+query+'%')) \
```

Comment:
we could actually construct this as an `or_` query instead of doing multiple queries. the only reason we had to do multiple queries in the other case is because we were querying from multiple tables

Reply:
Done